velox/exec/TopNRowNumber.cpp

```diff
@@ -432,37 +422,35 @@ RowVectorPtr TopNRowNumber::getOutputFromMemory() {
   }
 
   vector_size_t offset = 0;
-  if (remainingRowsInPartition_ > 0) {
-    auto& partition = currentPartition();
-    auto start = partition.rows.size() - remainingRowsInPartition_;
-    const auto numRows =
-        std::min<vector_size_t>(outputBatchSize_, remainingRowsInPartition_);
-    appendPartitionRows(partition, start, numRows, offset, rowNumbers);
-    offset += numRows;
-    remainingRowsInPartition_ -= numRows;
+  // If this is the first partition, then outputPartition_ is not set already.
+  if (!outputPartition_) {
+    outputPartition_ = nextPartition();
+    if (!outputPartition_) {
```

Comment:
This block must be at the beginning of the while loop as before. We must check after each `nextPartition()` call.
 The VELOX_CHECK(outputPartition_) on line 435 can be removed.
```
 if (!outputPartition_) {
    break; 
 }
```

Reply:
Its better to keep the check closer to the nextPartition() calls both up here and in the loop. So I moved the check at the end.

The previous code doing both checks made the loop more robust. But the current code organization is okay as well.

BTW, the TopNRowNumberFuzzer caught the check failures from before, so its good to know that its working well :)

Reply:
This refactor introduces an additional `if(!outputPartition_)` check in a while loop that could impact perf. That is my only concern.

Reply:
@majetideepak : You mean the one at line 431 ? The one on 427 matches the check in the loop before.

The loop does not iterate over each row... It is O(number of partitions) which is much smaller than the number of rows so I don't think it has the potential to introduce a perf degradation as such.